spec/filters/geoip_ecs_spec.rb

```diff
@@ -0,0 +1,203 @@
+          end
+        end
+
+      end
+    end
+
+    context "setup target field" do
+      let(:ip) { "8.8.8.8" }
+      let(:event) { LogStash::Event.new("message" => ip) }
+      let(:common_options) { {"source" => "message", "database" => CITYDB} }
+
+      context "ECS disabled" do
+        before do
+          allow_any_instance_of(described_class).to receive(:ecs_compatibility).and_return(:disabled)
+          plugin.register
+          plugin.filter(event)
+        end
+
+        context "`target` is unset" do
+          let(:options) { common_options }
+          it "should use 'geoip'" do
+            expect( event.get "[geoip][ip]" ).to eq ip
+          end
+        end
+
+        context "`target` is set" do
+          let(:target) { 'host' }
+          let(:options) { common_options.merge({"target" => target}) }
+          it "should use `target`" do
+            expect( event.get "[#{target}][ip]" ).to eq ip
+          end
+        end
+      end
+
+      context "ECS mode" do
+        before do
+          allow_any_instance_of(described_class).to receive(:ecs_compatibility).and_return(:v1)
+        end
+
+        context "`target` is unset" do
+
+          context "`source` end with [ip]" do
+            let(:event) { LogStash::Event.new("host" => {"ip" => ip}) }
+            let(:options) { common_options.merge({"source" => "[host][ip]"}) }
+
+            it "should use [host] as target" do
+              plugin.register
+              plugin.filter(event)
+              expect( event.get "[host][geo][country_iso_code]" ).to eq 'US'
+            end
+          end
+
+          context "`source` end with [ip] but `target` does not match ECS template" do
+            let(:event) { LogStash::Event.new("hostname" => {"ip" => ip}) }
+            let(:options) { common_options.merge({"source" => "[hostname][ip]"}) }
+
+            it "should use [hostname] as target with warning" do
+              expect(plugin.logger).to receive(:warn)
+              plugin.register
+              plugin.filter(event)
+              expect( event.get "[hostname][geo][country_iso_code]" ).to eq 'US'
+            end
+          end
+
+          context "`source` == [ip]" do
+            let(:event) { LogStash::Event.new("ip" => ip) }
+            let(:options) { common_options.merge({"source" => "[ip]"}) }
+
+            it "should raise error to require `target`" do
+              expect { plugin.register }.to raise_error LogStash::ConfigurationError
+            end
+          end
+
+          context "`source` not end with [ip]" do
+            let(:event) { LogStash::Event.new("host_ip" => ip) }
+            let(:options) { common_options.merge({"source" => "host_ip"}) }
+
+            it "should raise error to require `target`" do
+              expect { plugin.register }.to raise_error LogStash::ConfigurationError
```

Comment:
nit: same as above, assert error message contents

```suggestion
              expect { plugin.register }.to raise_error LogStash::ConfigurationError, /requires a `target`/
```